app/code/Magento/Sales/Model/Order/InvoiceCancel.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Sales\Model\Order;
+
+use Magento\Sales\Api\InvoiceCancelInterface;
+use Magento\Sales\Api\InvoiceRepositoryInterface;
+
+/**
+ * Class InvoiceCancel
+ */
+class InvoiceCancel implements InvoiceCancelInterface
+{
+
+    /**
+     * @var InvoiceRepositoryInterface
+     */
+    protected $invoiceRepository;
```

Comment:
Please avoid creating `protected` properties in the core codebase, as usage of inheritance is discouraged.